lxd/storage_pools.go

```diff
@@ -0,0 +1,352 @@
+	err = storagePoolValidateConfig(req.PoolName, req.PoolDriver, req.PoolConfig)
+	if err != nil {
+		return BadRequest(err)
+	}
+
+	// Create the database entry for the storage pool.
+	_, err = dbStoragePoolCreate(d.db, req.PoolName, req.PoolDriver, req.PoolConfig)
+	if err != nil {
+		return InternalError(fmt.Errorf("Error inserting %s into database: %s", req.PoolName, err))
+	}
+
+	// Define a function which reverts everything.  Defer this function
+	// so that it doesn't need to be explicitly called in every failing
+	// return path. Track whether or not we want to undo the changes
+	// using a closure.
+	tryUndo := true // CLEANER
+	defer func( /* CLEANER */ ) {
+		if tryUndo {
+			dbStoragePoolDelete(d.db, req.PoolName)
+		}
+	}()
+
+	s, err := storagePoolInit(d, req.PoolName)
+	if err != nil {
+		return InternalError(err)
+	}
+
+	err = s.StoragePoolCreate()
+	if err != nil {
+		return InternalError(err)
+	}
+	defer func( /* CLEANER */ ) {
+		if tryUndo {
+			s.StoragePoolDelete()
+		}
+	}()
+
+	// In case the storage pool config was changed during the pool creation,
+	// we need to update the database to reflect this change. This can e.g.
+	// happen, when we create a loop file image. This means we append ".img"
+	// to the path the user gave us and update the config in the storage
+	// callback. So diff the config here to see if something like this has
+	// happened.
+	postCreateConfig := s.GetStoragePoolWritable().PoolConfig
+	configDiff, _ := storageConfigDiff(req.PoolConfig, postCreateConfig)
+	if len(configDiff) > 0 {
+		// Create the database entry for the storage pool.
+		err = dbStoragePoolUpdate(d.db, req.PoolName, postCreateConfig)
+		if err != nil {
+			return InternalError(fmt.Errorf("Error inserting %s into database: %s", req.PoolName, err))
+		}
+	}
+
+	// Success, update the closure to mark that the changes should be kept.
+	tryUndo = false // CLEANER
+
+	return SyncResponseLocation(true, nil, fmt.Sprintf("/%s/storage-pools/%s", version.APIVersion, req.PoolName))
+}
+
+var storagePoolsCmd = Command{name: "storage-pools", get: storagePoolsGet, post: storagePoolsPost}
+
+// /1.0/storage-pools/{name}
+// Get a single storage pool.
+func storagePoolGet(d *Daemon, r *http.Request) Response {
+	poolName := mux.Vars(r)["name"]
+
+	// Get the existing storage pool.
+	poolID, pool, err := dbStoragePoolGet(d.db, poolName)
+	if err != nil {
+		return SmartError(err)
+	}
+
+	//  Get all users of the storage pool.
+	poolUsedBy, err := storagePoolUsedByGet(d.db, poolID, poolName)
+	if err != nil && err != NoSuchObjectError {
+		return SmartError(err)
+	}
+	pool.PoolUsedBy = poolUsedBy
+
+	etag := []interface{}{pool.PoolName, pool.PoolUsedBy, pool.PoolConfig}
+
+	return SyncResponseETag(true, &pool, etag)
+}
+
+// /1.0/storage-pools/{name}
+// Rename storage pool.
+func storagePoolPost(d *Daemon, r *http.Request) Response {
+	return BadRequest(fmt.Errorf("Storage pools can currently not be renamed."))
```

Comment:
If it's not implemented, just don't implement the API endpoint, that'll return a clean HTTP "Not implemented" rather than dealing with this in the code.